app/components/agent/Show.js

```diff
@@ -0,0 +1,267 @@
+import Panel from '../shared/Panel';
+import permissions from '../../lib/permissions';
+
+import AgentStopMutation from '../../mutations/AgentStop';
+
+const AGENT_REFRESH_INTERVAL = 5 * 1000;
+
+const CONNECTION_STATE_LABELS = {
+  'connected': 'Connected',
+  'disconnected': 'Disconnected',
+  'never_connected': 'Never Connected',
+  'lost': 'Lost Connection',
+  'stopped': 'Stopped',
+  'stopping': 'Stopping…'
+};
+
+class AgentShow extends React.Component {
+  static propTypes = {
+    agent: React.PropTypes.shape({
+      id: React.PropTypes.string.isRequired,
+      name: React.PropTypes.string.isRequired,
+      connectionState: React.PropTypes.string.isRequired,
+      permissions: React.PropTypes.shape({
+        agentStop: React.PropTypes.shape({
+          allowed: React.PropTypes.bool.isRequired
+        }).isRequired
+      }).isRequired
+    }),
+    relay: React.PropTypes.object.isRequired
+  };
+
+  state = {
+    stopping: false
+  };
+
+  componentDidMount() {
+    this._agentRefreshInterval = setInterval(this.fetchUpdatedData, AGENT_REFRESH_INTERVAL);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this._agentRefreshInterval);
+  }
+
+  fetchUpdatedData = () => {
+    this.props.relay.forceFetch(true);
+  };
+
+  renderExtraItem(title, content) {
+    return (
+      <li key={title}>
+        <strong className="black">{title}:</strong> {content}
+      </li>
+    );
+  }
+
+  renderExtras(agent) {
+    const extras = [];
+
+    if (agent.version) {
+      extras.push(this.renderExtraItem('Version', agent.version));
+    }
+
+    if (agent.hostname) {
+      extras.push(this.renderExtraItem('Hostname', agent.hostname));
+    }
+
+    if (agent.pid) {
+      extras.push(this.renderExtraItem('PID', agent.pid));
+    }
+
+    if (agent.ipAddress) {
+      extras.push(this.renderExtraItem('IP Address', agent.ipAddress));
+    }
+
+    if (agent.userAgent) {
+      extras.push(this.renderExtraItem('User Agent', agent.userAgent));
+    }
+
+    if (agent.operatingSystem) {
+      extras.push(this.renderExtraItem('OS', agent.operatingSystem));
+    }
+
+    if (agent.priority) {
+      extras.push(this.renderExtraItem('Priority', agent.priority));
+    }
+
+    if (agent.job) {
+      extras.push(this.renderExtraItem('Running', <a href="agent.job.path">{agent.job.projectName} - Build {agent.job.buildNumber} / <Emojify>{agent.job.name || agent.job.command}</Emojify></a>));
+    }
+
+    if (agent.connectedAt) {
+      extras.push(this.renderExtraItem(
+        'Connected',
+        <span>
+          <FriendlyTime value={agent.connectedAt} capitalized={true} />
+          {agent.pingedAt && agent.connectionState === 'connected' &&
+            <span> (last check-in was {<FriendlyTime value={agent.pingedAt} />})</span>
+          }
+        </span>
+      ));
+    }
+
+    if (agent.connectionState === 'disconnected') {
+      extras.push(this.renderExtraItem('Disconnected', <FriendlyTime value={agent.disconnectedAt} />));
```

Comment:
the label for `connectionState === 'lost'` is different; it could use the same label list, but I kept the behaviour the same as it currently is!